### PR TITLE
proxy: Demote errors from cplane request routines to debug

### PR DESCRIPTION
**Pull request by cloneable**
## Problem

Any errors from these async blocks are unconditionally logged at error level
even though we already handle such errors based on context.

## Summary of changes

* Log raw errors from creating and executing cplane requests at debug level.
* Inline macro calls to retain the correct callsite.

### CODE REVIEW

File: proxy/src/control_plane/client/neon.rs [hunk 1]
```diff
@@ -134,8 +134,8 @@ impl NeonControlPlaneClient {
                 project_id: body.project_id,
             })
         }
-        .map_err(crate::error::log_error)
-        .instrument(info_span!("http", id = request_id))
+        .inspect_err(|e| tracing::debug!(error = ?e))
+        .instrument(info_span!("do_get_auth_info", request_id))
```
Comment (conradludgate) [inline proxy/src/control_plane/client/neon.rs:138]: can we also remove the request_id? It's no longer used (we used to send it to cplane, but now we use session_id)

File: proxy/src/control_plane/client/neon.rs [hunk 2]
```diff
@@ -252,9 +252,9 @@ impl NeonControlPlaneClient {
 
             Ok(node)
         }
-        .map_err(crate::error::log_error)
+        .inspect_err(|e| tracing::debug!(error = ?e))
         // TODO: redo this span stuff
```
Comment (conradludgate) [inline proxy/src/control_plane/client/neon.rs:256]: remove?

File: proxy/src/control_plane/client/neon.rs (hunk 1 above)
Comment (cloneable) [reply proxy/src/control_plane/client/neon.rs:138]: yes, sure. The req id is not the sess id, so was wondering why we have it as "id" again.

File: proxy/src/control_plane/client/neon.rs (hunk 2 above)
Comment (cloneable) [reply proxy/src/control_plane/client/neon.rs]: Ok, I'm not sure what we discussed we wanted to do here. Maybe what I did here.

Comment (cloneable) [reply proxy/src/control_plane/client/neon.rs]: Done.

File: proxy/src/control_plane/client/neon.rs (hunk 1 above)
Comment (cloneable) [reply proxy/src/control_plane/client/neon.rs]: Done.

**Pull request merged**